homeassistant/components/sensor/sql.py

```diff
@@ -129,15 +129,17 @@ def update(self):
         finally:
             sess.close()
 
+        if not result.returns_rows or result.rowcount == 0:
```

Comment:
`returns_rows` -> I would expect this to be validated in the PLATFORM_SCHEMA by checking if the query starts with `SELECT` ?

Reply:
this is me being extra careful not to raise any exception :)

http://docs.sqlalchemy.org/en/latest/core/connections.html?highlight=returns_rows#sqlalchemy.engine.ResultProxy.returns_rows

How would I validade the query string using voluptuous ?
- Should I create a new "ensure_sql_query" in home-assistant/homeassistant/helpers/config_validation.py ?

Reply:
you can add a one off validator to this file and use that.

```
def validate_sql_select(value):
    if not value.lstrip().lower().startswith('select'):
        raise vol.Invalid('Only select queries allowed')
    return value

vol.Schema({
  'query': vol.All(cv.string, validate_sql_select)
})
```

Reply:
done!

Reply:
So now we can remove the `returns_rows` check ?

Reply:
whats is wrong with checking returns_rows ?